src/core/bittorrent/session.cpp

```diff
@@ -813,11 +813,21 @@ bool Session::deleteTorrent(const QString &hash, bool deleteLocalFiles)
 
     // Remove it from session
     if (deleteLocalFiles) {
-        QDir saveDir(torrent->actualSavePath());
-        if ((saveDir != QDir(m_defaultSavePath)) && (saveDir != QDir(m_tempPath))) {
-            m_savePathsToRemove[torrent->hash()] = saveDir.absolutePath();
-            qDebug() << "Save path to remove (async): " << saveDir.absolutePath();
+        if (torrent->filesCount() == 1) {
+            // handle special case for a single-file torrent like: "dir1/file1", "dir1/" should be removed
+            QString first_dir = torrent->firstFileSavePath();
+            QString tmp = first_dir;
+            while (QString(tmp).append("/") != torrent->actualSavePath()) {
+                first_dir = tmp;
+                tmp.truncate(tmp.lastIndexOf("/"));
+            }
+            if(first_dir != torrent->firstFileSavePath())
+                m_savePathsToRemove[torrent->hash()] = first_dir;
```

Comment:
Easier to cut off the first element from filePath(0) and add it to actualSavePath().